lib/client/db/profile.go

```diff
@@ -44,11 +44,32 @@ func Add(tc *client.TeleportClient, db tlsca.RouteToDatabase, clientProfile clie
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	// Postgres proxy listens on web proxy port while MySQL proxy listens on
-	// a separate port due to the specifics of the protocol.
-	host, port := tc.WebProxyHostPort()
-	if db.Protocol == defaults.ProtocolMySQL {
+	connectProfile, err := add(tc, db, clientProfile, profileFile)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if quiet {
+		return nil
+	}
+	switch db.Protocol {
```

Comment:
optional: if we add more protocols in the future, an interface (with per-protocol implementations) might be nicer than switches.

Reply:
I'll see if I can refactor this when I'm working on mongodb (our next protocol), though switches are simple enough I think.